src/main/java/com/AdCollection.java

```diff
@@ -115,47 +104,30 @@ public boolean edit(String id, Ad item) {
 
         Ad cloneAd = new Ad(currentAd);
 
-        Field[] fields = cloneAd.getClass().getDeclaredFields();
-        for (Field f : fields) {
-            switch (f.getName()) {
-                case "description": {
-                    cloneAd.setDescription(item.getDescription());
-                    break;
-                }
-                case "link": {
-                    cloneAd.setLink(item.getLink());
-                    break;
-                }
-                case "photoLink": {
-                    cloneAd.setPhotoLink(item.getPhotoLink());
-                    break;
-                }
-                case "hashTags": {
-                    cloneAd.setHashTags(item.getHashTags());
-                    break;
-                }
-                case "discount": {
-                    cloneAd.setDiscount(item.getDiscount());
-                    break;
-                }
-                case "validUntil": {
-                    cloneAd.setValidUntil(item.getValidUntil());
-                    break;
-                }
-                case "rating": {
-                    cloneAd.setRating(item.getRating());
-                    break;
-                }
-                case "reviews": {
-                    cloneAd.setReviews(item.getReviews());
-                    break;
-                }
-                default:
-                    break;
-            }
+        if (item.getDescription() != null) {

```

Comment:
Technically, all these condition checks are still redundant
You clone an ad.
Its fields are nulls
You try to assign a value of some other ad to a field of the current one. The value is null.
The result is still the same)

Coding is much about writing the minimal amount of code for achieving maximal result)